### PR TITLE
New Cli UI

### PR DESCRIPTION
**Pull request by araddon**
This PR changes the ui (flags, args, help-doc) to the CLI tool.  It moves to a _standard_ (ish) cli-command helper https://github.com/urfave/cli  .

It also:
* adds _format_ (table, json) to allow table based cli output which is nice/easy.
* deprecates some endpoints/functions which didn't appear to be as necessary for a base-tool.
  * provider's
  * segment size's
  * writing to files:    (use  `> file.json` instead)

Depends on:
https://github.com/lytics/go-lytics/pull/19

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,8 +19,7 @@ go get -u github.com/lytics/lytics
 
 ### Usage
 
-All examples use JQ https://stedolan.github.io/jq/ to 
-prettify the json output.
+All examples use JQ https://stedolan.github.io/jq/ to prettify the json output.
```
Comment (ashyablok) [inline README.md:23]: I suspect the read me needs a bit more of an update. Particularly in the example commands. For example I am seeing `segmentscan` in the README, and `segmentScan` in the new code.

File: cmds/main.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+		table.AddRow(row.Row()...)
+	}
+	fmt.Println(table.Render())
+}
+
+/*
+func writeToFile(file, data string) error {
+	err := ioutil.WriteFile(file, []byte(data), 0644)
+	return err
+}
```
Comment (ashyablok) [inline cmds/main.go:115]: Are we losing the ability to write output to a file? Do we ever plan to re-implement this?

File: cmds/segment.go [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+func (c *Cli) getSegmentAttributions(segments []string, limit int) (interface{}, error) {
+	attributions, err := c.Client.GetSegmentAttribution(segments)
+	if err != nil {
+		return nil, err
+	}
+
+	return attributions, nil
+}
+*/
```
Comment (ashyablok) [inline cmds/segment.go:108]: Would be nice to get a list of all the things we lose from the new implementation. That or some follow up tickets on what should be re-implemented post-merge.

File: cmds/account.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+				Usage:  "Show details of current authenticated account",
+				Action: accountShow,
+			},
+			{
+				Name:   "list",
+				Usage:  "List accounts",
+				Action: accountList,
+			},
+		},
+	})
```
Comment (ashyablok) [inline cmds/account.go:27]: This CLI package makes things so much nicer. 👏

**Review by mollyklecompte — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (araddon) [reply README.md:23]: I updated all of readme.

File: cmds/main.go (hunk 2 above)
Comment (araddon) [reply cmds/main.go:115]: No, no plan to reimplement.  not very unix-y:   ie this is the unix-y way to do it:
```sh
lytics segment scan last_2_hours > /tmp/users.json
```

File: cmds/segment.go (hunk 3 above)
Comment (araddon) [reply cmds/segment.go:108]: updated this pr to list them.

File: cmds/account.go (hunk 4 above)
Comment (araddon) [reply cmds/account.go:27]: yes/no its nice ui but kinda verbose to write.

**Pull request merged**